extensions/amp-experiment/0.1/amp-experiment.js

```diff
@@ -42,16 +43,16 @@ export class AmpExperiment extends AMP.BaseElement {
 
     const config = this.getConfig_();
     const results = Object.create(null);
-    this.experimentVariants = Promise.all(
-        Object.keys(config).map(experimentName => {
-          return allocateVariant(this.getWin(), config[experimentName])
-              .then(variantName => {
-                if (variantName) {
-                  results[experimentName] = variantName;
-                }
-              });
-        })).then(() => results);
+    const variants = Object.keys(config).map(experimentName => {
+      return allocateVariant(this.getWin(), config[experimentName])
+          .then(variantName => {
+            results[experimentName] = variantName;
+          });
+    });
+    this.experimentVariants = Promise.all(variants).then(() => results);
```

Comment:
is there a type declaration for `experimentVariants` ? and does it need to be public?